### PR TITLE
Adding hail phase changes and associated tests

### LINKED ISSUE #1743: Enhancement to phase chains to get hail-melt level (and ice-pellet level) and a hail-melted mask.
For the Hail from Global Data work, we need the `phase-change-level` and `phase-probability` CLIs and associated Plugins modified to calculate melting hail data, which will occur when the melting integral reaches 5000. K m.

Also consider how easy it would be to calculate an ice pellet level, that is, the point at which rain would freeze into ice pellets, this is a negative melting value which isn't currently calculated. We need to know  the equivalent integral for sub-zero temperatures in the case where snow HAS melted into either sleet or rain so that if a warm layer of air exists above a very cold layer, then ice pellets can be diagnosed instead of freezing rain. I'm not sure what this freezing threshold should be yet.

Acceptance criteria:
 * CLIs that can calculate a hail melting altitude and a hail melting mask, with acceptance tests showing this
 * Any thoughts on calculating an ice pellet level

### PR DESCRIPTION
**Pull request by mspelman07**
Closes #1743

Updates plugins to calculate the hail melt mask.

Acceptance test data on branch `mobt1743_hail_melt_level`

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by MoseleyS — CHANGES_REQUESTED** [review]
Looks good! I think I only found a few stray print statements to delete.

File: improver_tests/psychrometric_calculations/test_PhaseChangeLevel.py [hunk 1]
```diff
@@ -758,6 +769,38 @@ def test_sleet_rain_phase_change(self):
...
+                ]
+            )
+        )
+
+        expected = np.full_like(
+            self.expected_snow_sleet, fill_value=11.797252, dtype=np.float32
+        )
+        expected[:, 1:4, 1:4] = 1.0
+        expected[:, 2, 2] = 11.797252
+        print(result.data)
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver_tests/psychrometric_calculations/test_PhaseChangeLevel.py]: Done

File: improver_tests/psychrometric_calculations/precip_phase_probability/test_PrecipPhaseProbability.py [hunk 2]
```diff
@@ -149,6 +161,7 @@ def test_unit_conversion(self):
         result = self.plugin.process(self.cubes)
         expected = self.expected_template
         expected[2] = 1
+        print(expected)
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver_tests/psychrometric_calculations/precip_phase_probability/test_PrecipPhaseProbability.py]: Done

File: improver_tests/acceptance/test_phase_change_level.py [hunk 3]
```diff
@@ -74,4 +77,5 @@ def test_phase_change(tmp_path, phase_type, kgo_name, horiz_interp):
         output_path,
     ]
     run_cli(args)
+    print(kgo_path)
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver_tests/acceptance/test_phase_change_level.py]: Done

File: improver/psychrometric_calculations/precip_phase_probability.py [hunk 4]
```diff
@@ -107,7 +110,7 @@ def _extract_input_cubes(self, cubes: Union[CubeList, List[Cube]]) -> None:
             raise ValueError(
                 "Spatial coords mismatch between " f"{cubes[0]} and " f"{cubes[1]}"
             )
-
+        print(cubes)
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver/psychrometric_calculations/precip_phase_probability.py]: Done

**Review by MoseleyS — DISMISSED** [review]

**Review by Kat-90 — CHANGES_REQUESTED** [review]
This looks great. I've reviewed the code and asked for a couple minor changes, pulled the branch and run both unit tests and acceptance data. All looks fine. Happy to approve once those changes have been addressed. Thanks

File: improver/psychrometric_calculations/precip_phase_probability.py [hunk 5]
```diff
@@ -151,19 +160,23 @@ def process(self, cubes: Union[CubeList, List[Cube]]) -> Cube:
         the snow-sleet falling-level is supplied, this is the probability of
         snow at (or below) the surface. If the sleet-rain falling-level is
         supplied, this is the probability of rain at (or above) the surface.
+        If the hail-rain falling-level is
+        supplied, this is the probability of rain from hail at (or above) the surface.
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver/psychrometric_calculations/precip_phase_probability.py]: Done

File: improver_tests/psychrometric_calculations/test_PhaseChangeLevel.py [hunk 6]
```diff
@@ -75,6 +75,17 @@ def test_sleet_rain(self):
         self.assertEqual(plugin.phase_change_name, "rain_falling")
         self.assertEqual(plugin.grid_point_radius, 3)
 
+    def test_hail_rain(self):
+        """Test that the __init__ method configures the plugin as expected
+        for the sleet_rain phase change."""
```
[the remark this answers is not in the document]
Comment (mspelman07) [reply improver_tests/psychrometric_calculations/test_PhaseChangeLevel.py]: Done

**Review by Kat-90 — APPROVED** [review]
Thanks for the changes, happy to approve.

**Pull request merged**